scripts/lib/libvirt/compute-config.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+import argparse
+from string import Template
+import sys
+
+import libvirt
+import libvirt_helpers
+
+conn = libvirt.open("qemu:///system")
+if not conn:
+    print("Failed to open connection to the hypervisor")
+    sys.exit(1)
+
+parser = argparse.ArgumentParser(description="Create Compute Node Config.")
+parser.add_argument("cloud", type=str, help="Name of the cloud")
+parser.add_argument("nodecounter", type=str, help="Node Counter")
+parser.add_argument("macaddress", type=str, help="MAC Address")
+parser.add_argument("cephvolume", type=str, help="Ceph Volume")
+parser.add_argument("drbdvolume", type=str, help="DRBD Volume")
+parser.add_argument("computenodememory", type=str, help="Compute Node Memory")
+parser.add_argument("controllernodememory", type=str,
+                       help="Controller Node Memory")
+parser.add_argument("libvirttype", type=str, help="Libvirt Type")
+parser.add_argument("vcpus", type=str, help="VCPUs")
+parser.add_argument("emulator", type=str, help="Emulator")
+parser.add_argument("vdiskdir", type=str, help="Virtual Disk Directory")
+
+args = parser.parse_args()
+
+
+def main():
+    cloud = args.cloud
+    fout = open("/tmp/{0}-node{1}.xml".format(cloud, args.nodecounter), "w")
+    fin = libvirt_helpers.readfile("templates/compute-node.xml")
+    template = Template(fin)
+    cpuflags = libvirt_helpers.cpuflags()
+    libvirt_type = args.libvirttype
+    if libvirt_helpers.hypervisor_has_virtio(libvirt_type):
+        nicmodel = "e1000"
+    else:
+        nicmodel = "virtio"
+    if args.nodecounter == "1":
+        nodememory = args.computenodememory
+    else:
+        nodememory = args.controllernodememory
+
+    xml = template.substitute(cloud=args.cloud,
+                              nodecounter=args.nodecounter,
+                              nodememory=nodememory,
+                              vcpus=args.vcpus,
+                              cpuflags=cpuflags,
+                              emulator=args.emulator,
+                              vdisk_dir=args.vdiskdir,
+                              cephvolume=args.cephvolume,
+                              drbdvolume=args.drbdvolume,
+                              macaddress=args.macaddress,
+                              nicmodel=nicmodel)
+
+    #if libvirt_helpers.hypervisor_has_virtio(libvirt_type):
+        #sed -i -e "s/<target dev='vd\([^']*\)'
+        #bus='virtio'/<target dev='sd\1' bus='ide'/" $f
+
+    fout.write(xml)
+    fout.close()
+
+if __name__ == "__main__":
```

Comment:
Same in this file - currently this `__name__` check is useless because your CLI-handling code is in the wrong place.